### PR TITLE
Attention viz

### LINKED ISSUE #247: Attention Visualization
We can visualize the cross attention between the generative text decoder and the swin vision transformer encoder. This can be done by aggregating the attention scores for a given token on all attention maps, reshaping them into their 2D form. This can serve as some form of interpretability of the model and enable insights into its behavior. It would also be a nice addition to the final report.

### PR DESCRIPTION
**Pull request by Flova**
Closes #247

### CODE REVIEW

File: marimbabot_vision/scripts/attention_viz.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+import numpy as np
+import torch
+
+from PIL import Image
+from transformers import (DonutProcessor, VisionEncoderDecoderConfig,
+                          VisionEncoderDecoderModel)
+
+INPUT_FILE = './demo.png'
+MODEL = './model'
```
Comment (Juphex) [inline marimbabot_vision/scripts/attention_viz.py:19]: Maybe use the huggingface model path/repo, so, that we anyone can use this model for detection right away.

**Review by Juphex — COMMENTED** [review]
I reinstalled the requirements and got errors. I don't think this branch is the origin, but I want to clarify.

```
ERROR: torchmetrics 1.0.0 has requirement lightning-utilities>=0.7.0, but you'll have lightning-utilities 0.3.0 which is incompatible.
ERROR: tensorflow 2.13.0 has requirement typing-extensions<4.6.0,>=3.6.6, but you'll have typing-extensions 4.7.1 which is incompatible.
ERROR: sphinx 7.0.1 has requirement importlib-metadata>=4.8; python_version < "3.10", but you'll have importlib-metadata 1.5.0 which is incompatible.
ERROR: sphinx 7.0.1 has requirement requests>=2.25.0, but you'll have requests 2.22.0 which is incompatible.
ERROR: sphinx-rtd-theme 1.2.2 has requirement docutils<0.19, but you'll have docutils 0.20.1 which is incompatible.
ERROR: sphinx-rtd-theme 1.2.2 has requirement sphinx<7,>=1.6, but you'll have sphinx 7.0.1 which is incompatible.
ERROR: pytorch-lightning 1.8.1 has requirement PyYAML>=5.4, but you'll have pyyaml 5.3.1 which is incompatible.
ERROR: gruut 2.3.4 has requirement networkx<3.0.0,>=2.5.0, but you'll have networkx 3.1 which is incompatible.

```
Maybe we create a second requirements.txt that is for training only, so we get rid of these errors? If possible

**Comment by Flova** [comment]
> I reinstalled the requirements and got errors. I don't think this branch is the origin, but I want to clarify.
> 
> ```
> ERROR: torchmetrics 1.0.0 has requirement lightning-utilities>=0.7.0, but you'll have lightning-utilities 0.3.0 which is incompatible.
> ERROR: tensorflow 2.13.0 has requirement typing-extensions<4.6.0,>=3.6.6, but you'll have typing-extensions 4.7.1 which is incompatible.
> ERROR: sphinx 7.0.1 has requirement importlib-metadata>=4.8; python_version < "3.10", but you'll have importlib-metadata 1.5.0 which is incompatible.
> ERROR: sphinx 7.0.1 has requirement requests>=2.25.0, but you'll have requests 2.22.0 which is incompatible.
> ERROR: sphinx-rtd-theme 1.2.2 has requirement docutils<0.19, but you'll have docutils 0.20.1 which is incompatible.
> ERROR: sphinx-rtd-theme 1.2.2 has requirement sphinx<7,>=1.6, but you'll have sphinx 7.0.1 which is incompatible.
> ERROR: pytorch-lightning 1.8.1 has requirement PyYAML>=5.4, but you'll have pyyaml 5.3.1 which is incompatible.
> ERROR: gruut 2.3.4 has requirement networkx<3.0.0,>=2.5.0, but you'll have networkx 3.1 which is incompatible.
> ```
> 
> Maybe we create a second requirements.txt that is for training only, so we get rid of these errors? If possible

I don't think the training code is the issue tbh. also the libs are needed for inference at the moment. Also have a look at the ci, it doesn't throw these errors if I see it correctly.

**Review by Juphex — APPROVED** [review]
works

**Comment by Juphex** [comment]
@Flova you are right. The CI should be the baseline.

**Pull request merged**